spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkBatchQueryScan.java

```diff
@@ -291,17 +247,14 @@ public boolean equals(Object o) {
         readSchema().equals(that.readSchema()) && // compare Spark schemas to ignore field ids
         filterExpressions().toString().equals(that.filterExpressions().toString()) &&
         runtimeFilterExpressions.toString().equals(that.runtimeFilterExpressions.toString()) &&
-        Objects.equals(snapshotId, that.snapshotId) &&
-        Objects.equals(startSnapshotId, that.startSnapshotId) &&
-        Objects.equals(endSnapshotId, that.endSnapshotId) &&
-        Objects.equals(asOfTimestamp, that.asOfTimestamp);
+        ctx.equals(that.ctx);
   }
 
   @Override
   public int hashCode() {
     return Objects.hash(
-        table().name(), readSchema(), filterExpressions().toString(), runtimeFilterExpressions.toString(),
-        snapshotId, startSnapshotId, endSnapshotId, asOfTimestamp);
+        table().name(), readSchema(), filterExpressions().toString(),
+        runtimeFilterExpressions.toString(), ctx);
```

Comment:
Looks like `runtimeFilterExpressions.toString()` didn't need to move, which would keep the previous line from changing.

Reply:
Added context to that line.